doc/source/dev/database_session_management.md

```diff
@@ -0,0 +1,34 @@
+# How SQLAlchemy sessions are handled in Galaxy
+
+## Access to `sessionmaker`
+
+We have a central resource from which to grab SQLAlchemy's [sessionmaker](https://docs.sqlalchemy.org/en/20/orm/session_api.html#sqlalchemy.orm.sessionmaker), that is `app.model.session`.
+You can either use it directly as a proxy to the current session or call `app.model.session()` to get the proxied Session object, or use the context manager (e.g. `with app.model.session() as session:`), as detailed in the SQLAlchemy documentation.
+
+## `sessionmaker` scopes
+
+The default scope of the sessionmaker is [thread local](https://docs.sqlalchemy.org/en/20/orm/contextual.html#thread-local-scope). This means that accessing `sessionmaker` will return the same session in the same thread. This default scope can be changed if necessary.
+
+### WSGI, ASGI, Celery
+
+In the context of the webapp (WSGI and ASGI) there is a clearly delineated lifespan for a session, and that lifespan isn't tied to a thread. so we set the scope using `app.model.set_request_id(request_id)` and we close and remove the session at the end of the request with `app.model.unset_request_id(request_id)`. Similarly, we set the scope for each Celery task execution using `set_request_id(request_id)` and `unset_request_id`.
+
+By using this mechanism we are certain that web requests and celery tasks receive a new session object and that any resources held by the session are closed and rolled back.
```

Comment:
```suggestion
By using this mechanism we are certain that web requests and celery tasks receive a new session object and that any resources held by the session are released.
```
Just a suggestion: since we are rolling back only transactions, not all resources used by a session